### PR TITLE
Make vnode worker handle_work use cast instead of call.

### PR DESCRIPTION
**Pull request by kellymclaughlin**
Change `riak_core_vnode_worker:handle_work` to use `gen_server:cast` instead of `gen_server:call`. This addresses occasional timeouts that can cause the vnode worker pool to crash.

### CODE REVIEW

File: src/riak_core_vnode_worker.erl [hunk 1]
```diff
@@ -42,16 +42,18 @@ start_link(Args) ->
...
 
 init([Module, VNodeIndex, WorkerArgs, WorkerProps]) ->
     {ok, WorkerState} = Module:init_worker(VNodeIndex, WorkerArgs, WorkerProps),
     {ok, #state{module=Module, modstate=WorkerState}}.
 
-handle_call({work, Pool, Work, WorkFrom}, {Pid, _} = From, #state{module = Mod,
-        modstate = ModState} = State) ->
-    gen_server:reply(From, ok), %% unblock the caller
+handle_call(_Event, _From, State) ->
+    {reply, ok, State}.
```
Comment (reiddraper) [inline src/riak_core_vnode_worker.erl:57]: Should we be expecting to get messages here? Maybe a lager debug msg if do?

File: src/riak_core_vnode_worker.erl [hunk 2]
```diff
@@ -42,16 +42,18 @@ start_link(Args) ->
     [VNodeIndex, WorkerArgs, WorkerProps] = proplists:get_value(worker_args, Args),
     gen_server:start_link(?MODULE, [WorkerMod, VNodeIndex, WorkerArgs, WorkerProps], []).
 
-handle_work(Pid, Pool, Work, From) ->
-    gen_server:call(Pid, {work, Pool, Work, From}).
+handle_work(Worker, Pool, Work, From, Caller) ->
```
Comment (reiddraper) [inline src/riak_core_vnode_worker.erl:48]: Is caller ever not `self()`?

Comment (reiddraper) [inline src/riak_core_vnode_worker.erl:48]: Maybe another 4 arity version

``` erlang
handle_work(Worker, Pool, Work, From) ->
    handle_worker(Worker, Pool, Work, From, self()).
```

**Comment by kellymclaughlin** [comment]
Comments addressed by [205e1dd](https://github.com/basho/riak_core/commit/205e1dd7b10920d017beda6186bbe316ec551c87).

**Comment by reiddraper** [comment]
merge it! nice work

**Comment by kellymclaughlin** [comment]
Rebased from master and squashed commits.

**Pull request merged**